Allow build with Gcc > 2 to be done

--- xorg.cf	2022-05-03 13:01:22.284734900 +0000
+++ my_xorg.cf	2022-05-03 13:00:51.334734900 +0000
@@ -1567,10 +1567,16 @@
 # endif
 #endif
 
-#if HasGcc2 && defined(AMD64Architecture)
-# ifndef DefaultGcc2AMD64Opt
+/*
+ * linux.cf has DefaultCDebugFlags to be DefaultGcc2AMD64Opt
+ * so if compiling with Gcc, we have to at least define DefaultGcc2AMD64Opt
+ * Otherwise, no macro substitution for DefaultGcc2AMD64Opt -> build failure
+ */
+#ifndef DefaultGcc2AMD64Opt && defined(AMD64Architecture)
+# if defined(HasGcc2) || defined(HasGcc3)
 #  define DefaultGcc2AMD64Opt -O2 -fno-strength-reduce GccAliasingArgs
-# endif
+# elif defined(HasGcc)
+#  define DefaultGcc2AMD64Opt
 #endif
 
 #if HasGcc2 && defined(AlphaArchitecture)
